dali/base/dadfs.cpp

```diff
@@ -8228,12 +8230,19 @@ void CDistributedFileDirectory::addEntry(CDfsLogicalFileName &dlfn,IPropertyTree
     IPropertyTree *prev = getNamedPropTree(sroot,superfile?queryDfsXmlBranchName(DXB_SuperFile):queryDfsXmlBranchName(DXB_File),"@name",tail.str(),false);
     if (!prev) // check super/file doesn't exist
         prev = getNamedPropTree(sroot,superfile?queryDfsXmlBranchName(DXB_File):queryDfsXmlBranchName(DXB_SuperFile),"@name",tail.str(),false);
-    if (prev!=NULL) {
+    if (prev!=NULL)
+    {
+        IDFS_Exception *e = nullptr;
+        Owned<IPropertyTreeIterator> iter = prev->getElements("SuperOwner");
+        if (iter->first() && !ignoreexists)
+            e = new CDFS_Exception(DFSERR_OwnedBySuperfile,dlfn.get());
         prev->Release();
         root->Release();
         if (ignoreexists)
             return;
```

Comment:
minor: would be tidier if prev and root were Owned. Then you could move 'if (ignoreexists) return' up, remove the Release() calls and throw exception directly (without temp. 'e').